test/inc/zoo/variant.h

```diff
@@ -73,8 +90,24 @@ template<typename Visitor, typename Var>
 auto visit(Visitor &&visitor, Var &&var) ->
     decltype(visitor(get<0>(std::forward<Var>(var))));
 
+namespace detail {
+
+    template<typename T, typename Enable = void>
+    struct DestroyInvoker {
+    };
+
+    template<template<typename...> class Var, typename... Ts>
+    struct DestroyInvoker<Var<Ts...>,
+            std::enable_if_t<!(std::is_trivially_destructible_v<Ts> && ...)>> {
+        ~DestroyInvoker() {
+            static_cast<Var<Ts...>*>(this)->destroy();
```

Comment:
This line is "undefined behavior", because this is a base class, the `Variant` instance has already been destroyed, but this wants to use it again.  It is a violation of the abstract machine.  This is a technical thing, but not desirable